connectivity/check/wait.go

```diff
@@ -0,0 +1,250 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package check
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/cilium/cilium-cli/defaults"
+	"github.com/cilium/cilium-cli/k8s"
+)
+
+const (
+	LongTimeout  = 5 * time.Minute
+	ShortTimeout = 30 * time.Second
+
+	PollInterval = 1 * time.Second
+)
+
+// WaitForDeployment waits until the specified deployment becomes ready.
+func WaitForDeployment(ctx context.Context, log Logger, client *k8s.Client, namespace string, name string) error {
+	log.Logf("⌛ [%s] Waiting for deployment %s/%s to become ready...", client.ClusterName(), namespace, name)
+
+	ctx, cancel := context.WithTimeout(ctx, LongTimeout)
+	defer cancel()
+	for {
+		err := client.CheckDeploymentStatus(ctx, namespace, name)
+		if err == nil {
+			return nil
+		}
+
+		log.Debugf("[%s] Deployment %s/%s is not yet ready: %s", client.ClusterName(), namespace, name, err)
+
+		select {
+		case <-time.After(PollInterval):
+		case <-ctx.Done():
+			return fmt.Errorf("timeout reached waiting for deployment %s/%s to become ready (last error: %w)",
+				namespace, name, err)
+		}
+	}
+}
+
+// WaitForCiliumEndpoint waits until the specified cilium endpoint gets created.
+func WaitForCiliumEndpoint(ctx context.Context, log Logger, client *k8s.Client, namespace, name string) error {
+	log.Logf("⌛ [%s] Waiting for CiliumEndpoint for pod %s/%s to appear...", client.ClusterName(), namespace, name)
+
+	ctx, cancel := context.WithTimeout(ctx, ShortTimeout)
+	defer cancel()
+	for {
+		_, err := client.GetCiliumEndpoint(ctx, namespace, name, metav1.GetOptions{})
```

Comment:
nit: one liner check

Reply:
I don't think that this can be refactored to the be a one liner check, because we return if the error is nil, and we access it otherwise. The same for the ones below.